Assets/Plugins/Support/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,123 @@
+using Support;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using NaughtyAttributes;
+namespace Support { 
+    public class AudioManager : MonoSingleton<AudioManager>
+    {
+        //getting resources
+        [Foldout("Source")]
+        [SerializeField]
+        [ReorderableList]
+        private Sound[] sounds;
+
+        [Foldout("Source")]
+        [SerializeField]
+        [ReorderableList]
+
+        private Sound[] music;
+        [Foldout("Source")]
+        [SerializeField]
+        [ReorderableList]
+        private Sound[] dialogs;
+        //Dictionary - fetch resources from Arrays to simplisify choosing a sound
+        private Dictionary<string,Sound> _sounds = new Dictionary<string, Sound>();
+        private Dictionary<string,Sound> _music = new Dictionary<string, Sound>();
+        private Dictionary<string,Sound> _dialogs = new Dictionary<string, Sound>();
+
+        private void Awake()
+        {
+            CopySoundsListIntoDictionary(dialogs, _dialogs);
+            CopySoundsListIntoDictionary(music,_music );
+            CopySoundsListIntoDictionary(sounds,_sounds );
+        }
+
+        #region PRIVATE_METHODS
+        private void CopySoundsListIntoDictionary(Sound[] arr, Dictionary<string, Sound> dic)
+        {
+            foreach(Sound s in arr)
+            {
+                s.Source = gameObject.AddComponent<AudioSource>();
+                s.Source.clip = s.Clip;
+                s.Source.pitch = s.Pitch;
+                s.Source.volume = s.Volume;
+                s.Source.loop = s.Loop;
+                s.Source.playOnAwake = false;
+                dic.Add(s.Name, s);
+            }
+        }
+        
+        private Sound GetRawSound(Dictionary<string, Sound> arr,string s)
```

Comment:
Creating such shortcuts may be expensive as it is one additional **function call**, you can replace that method with one line:
`var res = arr.TryGetValue(s, out result) ? result : null;`